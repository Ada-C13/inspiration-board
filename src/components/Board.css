.board {
  display: flex;
  flex-wrap: wrap;
  /* centers board */
  margin-left: auto;
  margin-right: auto;
  align-items: center;
  justify-content: center;
}

.validation-errors-display {
  text-align: center;
  margin-bottom: 2em;
}

.validation-errors-display__list {
  list-style-type: none;
  padding: 0;
}
