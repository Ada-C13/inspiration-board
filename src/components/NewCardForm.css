/* The Modal (background) */

.new-card-form {
  width: 100%;
  padding-bottom: 4rem;
}

.new-card-form__header {
  text-align: center;
  text-transform: uppercase;
}

.new-card-form__form {
  width: 50%;
  margin: auto;
  font-size: 1.5em;
  display: grid;
  grid-template-columns: [labels] auto [controls] 1fr;
  grid-auto-flow: row;
}

.new-card-form__form-label  {
  grid-column: labels;
  grid-row: auto;
}

.new-card-form__form-select,
.new-card-form__form-textarea,
.new-card-form__form-button {
  font-family: 'Raleway', sans-serif;
  font-size: 1.25em;
  grid-column: controls;
  grid-row: auto;
  margin-bottom: .8em;
}

.new-card-form__form-button {
  background-color: inherit;
  border: 1px solid black;
  font-size: 1em;
}
