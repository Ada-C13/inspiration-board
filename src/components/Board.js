import React, { Component, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';

import './Board.css';
import Card from './Card';
import NewCardForm from './NewCardForm';
import CARD_DATA from '../data/card-data.json';

const Board = (props) => {
  const BASE_URL = `${props.url}${props.boardName}/cards`

  const[cards, setCards] = useState([]);
  const[errorMessage, setErrorMessage] = useState(null);

  useEffect(() => {
    axios.get(BASE_URL)
    .then((response) => {
      const apiData = response.data;
      console.log(apiData);

      const cardObjects = apiData.map((cardWrapper, i) => {
        return {
          id: cardWrapper.card.id,
          text: cardWrapper.card.text,
          emoji: cardWrapper.card.emoji
        }
      });
      setCards(cardObjects);
      })
      .catch((error) => {
        setErrorMessage(error.message);
      });
  }, []);

  // Wave 3
  const deleteCard = (id) => {
    axios.delete(`https://inspiration-board.herokuapp.com/cards/${id}`)
    .then((response) => {
      const newCards = cards.filter(card => card.id !== id);
      setCards(newCards);
    })
    .catch((error) => {
      setErrorMessage(error.message);
    })
  };

  // Wave 3
  const addCard = (cardInfo) => {
    const nextId = Math.max(...cards.map(card => card.id)) + 1;

    axios.post(BASE_URL, cardInfo)
    .then((response) => {
      const newCards = [...cards];
      const newCard = {
        id: nextId,
        ...cardInfo
      };
      newCards.unshift(newCard);
      setCards(newCards);
    })
    .catch((error) => {
      setErrorMessage(error.message);
    })
  }

  
  const cardComponents = cards.map((card, i) => {
    return (
      <Card 
      id={card.id}
      key={i}
      text={card.text}
      emoji={card.emoji}
      deleteCardCallback={deleteCard}
      />
    );
  });

  return (
    <div>
      {errorMessage && <div><h2>{errorMessage}</h2></div>}
      {cardComponents}
      <NewCardForm 
        addCardCallback={addCard}
      />
    </div>
  )
};


Board.propTypes = {
  url: PropTypes.string.isRequired,
  boardName: PropTypes.string.isRequired
};

export default Board;
