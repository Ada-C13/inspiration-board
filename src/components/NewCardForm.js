import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

const NewCardForm =(props) => {

  const emojiDropdown = () => {
    const emojiList = EMOJI_LIST.map(emoji=>{
      return  (<option value={emoji}>{emoji}</option>);
    });
    return emojiList;
  };



  return(

    <form className="NewCardForm" onSubmit={props.onSubmitCard}>
    <label>Card Text:</label>
     <input name = "cardtext" type="text" onChange={props.onChangeCard}/>
    <label>Emoji Input:</label>
     <select  name="cardemoji" onChange={props.onChangeCard}>
      {emojiDropdown()}
    </select>
    <input
        type="submit"
        value="Submit Card"
      />
    </form>
  );
};

export default NewCardForm;