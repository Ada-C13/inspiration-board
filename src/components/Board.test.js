import React from 'react';
import { render, cleanup } from '@testing-library/react'
import Board from './Board';
import NewCardForm from './NewCardForm';

describe ('Board', () => {
  // test('renders one card of off the Board', () => {
  //     const { getByText } = render(<Card />);
  //     const linkElement = getByText(/nap/i);
  //     expect(linkElement).toBeInTheDocument();
  // });
  
  test('will render the NewCardForm properly', () => {
    // Arrange-Act
    const { asFragment } = render(
      <NewCardForm
        addCardCallback={() => { }}
      />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
})



















