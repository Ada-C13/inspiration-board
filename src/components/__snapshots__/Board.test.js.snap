// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="validation-errors-display"
    />
    <div
      class="board"
    >
      <form
        class="new-card-form"
      >
        <div
          class="new-card-form__header"
        />
        <div
          class="new-card-form__form"
        >
          <label
            class="new-card-form__form-label"
            for="text"
          >
            text
          </label>
          <input
            class="new-card-form__form-textarea"
            name="text"
            placeholder="text"
            type="text"
            value=""
          />
          <label
            class="new-card-form__form-label"
            for="emoji"
          >
            emoji
          </label>
          <select
            class="new-card-form__form-select"
            name="emoji"
          >
            <option
              value=""
            >
              Select an Emoji!
            </option>
            <option
              value="heart_eyes"
            >
              😍
            </option>
            <option
              value="smiling_face_with_three_hearts"
            >
              🥰
            </option>
            <option
              value="zany"
            >
              🤪
            </option>
            <option
              value="eyes"
            >
              👀
            </option>
            <option
              value="sparkles"
            >
              ✨
            </option>
            <option
              value="sunflower"
            >
              🌻
            </option>
            <option
              value="seedling"
            >
              🌱
            </option>
            <option
              value="cherry_blossom"
            >
              🌸
            </option>
            <option
              value="beer"
            >
              🍺
            </option>
            <option
              value="clap"
            >
              👏
            </option>
            <option
              value="heart"
            >
              ❤️
            </option>
            <option
              value="sparkling_heart"
            >
              💖
            </option>
            <option
              value="heart_eyes_cat"
            >
              😻
            </option>
            <option
              value="dog"
            >
              🐶
            </option>
            <option
              value="sob"
            >
              😭
            </option>
            <option
              value="fu"
            >
              🖕
            </option>
            <option
              value="hugs"
            >
              🤗
            </option>
            <option
              value="partying"
            >
              🥳
            </option>
            <option
              value="star_struck"
            >
              🤩
            </option>
            <option
              value="sunglasses"
            >
              😎
            </option>
            <option
              value="space_invader"
            >
              👾
            </option>
            <option
              value="brain"
            >
              🧠
            </option>
            <option
              value="takeout_box"
            >
              🥡
            </option>
            <option
              value="nerd_face"
            >
              🤓
            </option>
            <option
              value="fire"
            >
              🔥
            </option>
            <option
              value="rainbow"
            >
              🌈
            </option>
            <option
              value="mask"
            >
              😷
            </option>
          </select>
          <input
            class="new-card-form__form-button"
            type="submit"
            value="Post a Card"
          />
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
